components/Header.tsx

```diff
@@ -0,0 +1,49 @@
+'use client';
+
+import { useSupabase } from '@/app/supabase-provider';
+import Link from 'next/link';
+import { useEffect, useState } from 'react';
+import type { Database } from '@/lib/database.types';
+
+export default function Header() {
+  const [profile, setProfile] = useState<Database['public']['Tables']['profiles']['Row']>();
+  const [loggedIn, setLoggedIn] = useState<boolean>(false);
```

Comment:
Isn't this a piece of derived state? That is - the loggedIn variable is identical to `Boolean(profile)`, right? If so, you can get away with just one `useState()` hook

Reply:
True, I'll make those changes